hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/StreamWriteOperatorCoordinator.java

```diff
@@ -305,6 +306,13 @@ public void subtaskFailed(int i, @Nullable Throwable throwable) {
     // reset the event
     this.eventBuffer[i] = null;
     LOG.warn("Reset the event for task [" + i + "]", throwable);
+    if (Arrays.stream(this.eventBuffer).allMatch(event -> event == null)) {
```

Comment:
Can we roll back this change now, not sure if it is right.

Reply:
In `StreamWriteOperatorCoordinator#notifyCheckpointAborted`,
we can add another check condition:
`checkpointId == this.checkpointId && !WriteMetadataEvent.BOOTSTRAP_INSTANT.equals(this.instant)`
to make sure not abort the instant during first time startup for rollback